src/StaticMock/Mock.cs

```diff
@@ -133,6 +133,165 @@ public static void SetupDefault(Type type, string methodName, Action action)
             MockHelper.SetupDefault(methodToReplace, action);
         }
 
+        public static IFuncMockService Setup(Type type, string methodName, BindingFlags bindingFlags, Action action)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            if (methodName == null)
+            {
+                throw new ArgumentNullException(nameof(methodName));
+            }
+
+            string[] values = bindingFlags.ToString().Split(new[] { ", " }, StringSplitOptions.None);
+            foreach (string value in values)
+            {
+                if (!Enum.IsDefined(typeof(BindingFlags), value))
+                {
+                    throw new ArgumentException(nameof(bindingFlags));
+                }
+            }
```

Comment:
This is unnecessary check

Reply:
Why this is not a successful check, please let me know　((´∀｀))

Reply:
this is unnecessary check because C# is strong typing language and compiler won't let you to put flags that isn't existing in BindingFlags enum. This check has the meaning if you put array of string as parameter and want that this array of string contains only values of this enum.

Reply:
I see what you mean. Thank you very much. However, the following situations may exist, and there will be no errors when compiling. 666 is not an enumeration type. When it is passed as a parameter after type conversion, do we need to check it？
BindingFlags　bindingFlags =　BindingFlags.Static | BindingFlags.NonPublic | (BindingFlags) 666 ;

Reply:
I think we don't need to check this case. Users how will do it like this will be responsible for this themselves.

Reply:
OK, thank you very much. Hope more guidance in the future